### ISSUE TITLE
Tracking Management Commands

### ISSUE THREAD

**Issue by gabriel-v** [Tijani-Dia/dj-tracker#24]
How I would go about to track management commands and their queries too?

I'm interested of tracking the queries made by scripts, Celery workers, etc. All of these are Django standard management commands.

Do I need to run some kind of internal tracker in a parallel thread?

Here's a benchmark that I would like to see show up in the interface:

```
/manage.py shell -c "from django.contrib.auth.models import User; [print(list(x.username for x in User.objects.all())) for _ in range(1000)]"
```

This would show 1000 queries (preferrably under a separate list in the UI) that each pulls the whole User table and only gets the username.

**Comment by Tijani-Dia** [comment]
Hey @gabriel-v, a way to make this happen is to call `tracker.start()` right before doing the queries, and optionally `tracker.stop()`.
That's how the [middleware does it](https://github.com/Tijani-Dia/dj-tracker/blob/main/src/dj_tracker/middleware.py#L5) but also the [custom test runner](https://github.com/Tijani-Dia/dj-tracker/blob/main/src/dj_tracker/test.py#L16-L20) (which you could use to track queries for your test suite).

So, I think:
```
/manage.py shell -c "from django.contrib.auth.models import User; from dj_tracker import tracker; tracker.start(); [print(list(x.username for x in User.objects.all())) for _ in range(1000)]; tracker.stop()"
```
should do it. Please let me know if it doesn't.

In this case, all the queries will be shown under the `DummyRequest` section which is more of a placeholder. We'll need a better way to distinguish where the queries come from when running the tracker outside a request.

> This would show 1000 queries for the shell command (preferrably under a separate list in the UI).

That'd definitely be great.

**Comment by gabriel-v** [comment]
Ah, DummyRequest! They're all there, thanks!

Another couple of questions:
- how come they don't show up in the "Queries" main section? 
- the " Model attributes accessed " field does not expand when "Show" is clicked

![Screenshot from 2023-06-22 22-56-28](https://github.com/Tijani-Dia/dj-tracker/assets/7493327/f513342c-6060-4ce3-9fe3-47a2b9c7cd36)

**Comment by Tijani-Dia** [comment]
> how come they don't show up in the "Queries" main section?

You may be referring To [this line](https://github.com/Tijani-Dia/dj-tracker/blob/main/src/dj_tracker/views.py#L108) where they're filtered out from the queryset. I feel it adds a lot of noise as it associate all queries outside a request in there and I'd have thousands of them in some situations and it's hard to find anything useful with that much data. As I mentioned earlier, we'll need a more granular way to group those queries e.g say these queries come from this test suite or from this task or just find a way to label them.

> the " Model attributes accessed " field does not expand when "Show" is clicked

It means no model attributes were accessed in the sense of custom attributes or methods and such. In the script above, it seems that you're only accessing the `username` field which should be shown in the `Field stats` section when you click on the query link.

**Comment by gabriel-v** [comment]
> should be shown in the Field stats section when you click on the query link.

Yes, on real code it works perfectly. The table header is also a link - it's obvious now. Thanks!

> You may be referring to [this line](https://github.com/Tijani-Dia/dj-tracker/blob/main/src/dj_tracker/views.py#L108)

Ah, so I could patch the tracker some way after starting it, to overwrite the request path? I could set some dummy path for it, based on the management command ID & args...
